### PR TITLE
Refactor Builder API before documentation

### PR DESCRIPTION
**Pull request by FabienLelaquais**
Addresses [doc#533](https://github.com/Avaiga/taipy-doc/issues/533).

### CODE REVIEW

File: src/taipy/gui/_renderers/__init__.py [hunk 1]
```diff
@@ -32,27 +23,28 @@
...
         if content is None:
             raise ValueError("'content' argument is missing for class '_Renderer'")
         self._content = ""
-        self._base_element: t.Optional[ElementApi] = None
+        self._base_element: t.Optional["_Element"] = None
         self._filepath = ""
         if isinstance(content, str):
             self.__process_content(content)
-        elif isinstance(content, ElementApi):
+        elif any(filter(lambda c: c.__name__.endswith("_Element"), type(content).__mro__)):
```
Comment (FredLL-Avaiga) [inline src/taipy/gui/_renderers/__init__.py]: a bit scary

**Review by FredLL-Avaiga — APPROVED** [review]
Quite a change

File: src/taipy/gui/_renderers/__init__.py (hunk 1 above)
Comment (FabienLelaquais) [reply src/taipy/gui/_renderers/__init__.py]: Indeed
Let's fix that if you want.

Comment (FabienLelaquais) [reply src/taipy/gui/_renderers/__init__.py]: So far, we failed...
@dinhlongviolin1, any idea?

Comment (FabienLelaquais) [reply src/taipy/gui/_renderers/__init__.py]: Fixed!

**Review by dinhlongviolin1 — APPROVED** [review]
LGTM

**Pull request merged**